doc/encryption: improve consistency between ipsec and wireguard guides (#15965)
cilium-team/cilium: Documentation/gettingstarted/encryption-wireguard.rst

is then used by other nodes to decrypt and encrypt traffic from and to
Cilium-managed endpoints running on that node.

Packets destined to the same node they were sent out of are not encrypted.

> This is a intended behavior as it doesn't provide any benefits because the

Reviewer: ```suggestion
This is an intended behavior as it would not provide any benefits because the
```

Author: I copy pasted this sentence from the ipsec page but I'm happy to update both instances.
